pandas/core/arrays/integer.py

```diff
@@ -218,6 +222,7 @@ def coerce_to_array(values, dtype, mask=None, copy=False):
             "mixed-integer",
             "integer-na",
             "mixed-integer-float",
+            "boolean",
```

Comment:
why is this added?

Reply:
Otherwise it says, TypeError: object cannot be converted to an IntegerDtype.

Reply:
what says this?

Reply:
Actually, when this error came out, @jorisvandenbossche told me to add "boolean" as inferred dtype.  https://github.com/pandas-dev/pandas/pull/30282#issuecomment-566005532.
Honestly I also don't understand the reason behind that, but it just got fixed by adding "boolean" as inferred dtype there.

Reply:
Actually, when this error came out, @jorisvandenbossche told me to add "boolean" as inferred dtype. #30282 (comment).
Honestly, I also don't know the reason behind it. But it just got fixed by doing so.